app/Helpers/database/hash.php

```diff
@@ -1,142 +1,91 @@
 <?php
 
-use App\Community\Enums\ArticleType;
+use App\Models\Game;
+use App\Models\GameHash;
 
-function getMD5List(int $consoleID): array
+function getMD5List(int $consoleId): array
 {
-    $retVal = [];
-
-    $whereClause = "";
-    if ($consoleID > 0) {
-        $whereClause = "WHERE gd.ConsoleID = $consoleID ";
-    }
-
-    $query = "SELECT MD5, GameID
-              FROM GameHashLibrary AS ghl
-              LEFT JOIN GameData AS gd ON gd.ID = ghl.GameID
-              $whereClause
-              ORDER BY GameID ASC";
-
-    foreach (legacyDbFetchAll($query) as $nextData) {
-        $nextData['GameID'] = (int) $nextData['GameID'];
-        $retVal[$nextData['MD5']] = $nextData['GameID'];
-    }
-
-    return $retVal;
+    $query = GameHash::query()
+        ->select('game_hashes.md5', 'game_hashes.game_id')
+        ->leftJoin('GameData as gd', 'gd.ID', '=', 'game_hashes.game_id')
+        ->when($consoleId > 0, function ($q) use ($consoleId) {
+            $q->where('gd.ConsoleID', $consoleId);
+        })
+        ->orderBy('game_hashes.game_id', 'asc');
+
+    return $query->pluck('game_id', 'md5')->toArray();
 }
 
-function getHashListByGameID(int $gameID): array
+function getHashListByGameID(int $gameId): array
 {
-    if ($gameID < 1) {
+    if ($gameId < 1) {
         return [];
     }
 
-    $query = "SELECT MD5 AS Hash, Name, Labels, User
-              FROM GameHashLibrary
-              WHERE GameID = $gameID
-              ORDER BY Name, Hash";
-
-    $retVal = [];
-    $dbResult = s_mysql_query($query);
-    if ($dbResult !== false) {
-        while ($nextData = mysqli_fetch_assoc($dbResult)) {
-            $retVal[] = $nextData;
-        }
+    $game = Game::find($gameId);
+    if (!$game) {
+        return [];
     }
 
-    return $retVal;
+    $hashes = $game->hashes()
+        ->with('user')
+        ->select('md5', 'name', 'labels', 'user_id')
+        ->orderBy('name')
+        ->orderBy('md5')
+        ->get()
+        ->map(function ($hash) {
+            return [
+                'Hash' => $hash->md5,
+                'Name' => $hash->name,
+                'Labels' => $hash->labels,
+                'User' => $hash->user ? $hash->user->User : null,
```

Comment:
Can be shortened to `'User' => $hash->user?->User,`

Reply:
Thanks, adjusted in latest 👍